conda/core/link.py

```diff
@@ -193,6 +194,7 @@ class PrefixActionGroup(NamedTuple):
     make_menu_action_groups: Iterable[ActionGroup]
     entry_point_action_groups: Iterable[ActionGroup]
     prefix_record_groups: Iterable[ActionGroup]
+    final_transaction_groups: Iterable[ActionGroup]
```

Comment:
Unfortunately, namedtuple args are positional, so this is an API change that needs to go through the deprecation cycle.

What if we create a new `PrefixActionGroup` namedtuple with the new arguments, and deprecate the old one? e.g. `PrefixActionGroupV2` or similar? Even better if the new object is a dataclass so it doesn't suffer from this problem in the future?

Reply:
Ack you're right! Wow, the `PrefixActionGroup` feels like such an implementation detail I'm surprised it's not private. Deprecation, and the dataclass was nearly a drop-in replacement :rocket: